core/server/master/src/main/java/alluxio/master/block/BlockMaster.java

```diff
@@ -174,10 +175,12 @@ void workerRegister(long workerId, List<String> storageTiers,
    * @param usedBytesOnTiers a mapping from tier alias to the used bytes
    * @param removedBlockIds a list of block ids removed from this worker
    * @param addedBlocksOnTiers a mapping from tier alias to the added blocks
+   * @param options additional block heartbeat options
    * @return an optional command for the worker to execute
    */
   Command workerHeartbeat(long workerId, Map<String, Long> usedBytesOnTiers,
-      List<Long> removedBlockIds, Map<String, List<Long>> addedBlocksOnTiers);
+      List<Long> removedBlockIds, Map<String, List<Long>> addedBlocksOnTiers,
+      BlockHeartbeatTOptions options);
```

Comment:
The convention is to use the wire type instead of the thrift type in the master API